fetch/data-urls/resources/data-urls.json

```diff
@@ -0,0 +1,175 @@
+[
+  ["data://test/,X",
+   "",
+   [88]],
+  ["data://test:test/,X",
+   null],
+  ["data:,X",
+   "",
+   [88]],
+  ["data:,X#X",
+   "",
+   [88]],
+  ["data:,%FF",
+   "",
+   [255]],
+  ["data:text/plain,X",
+   "text/plain",
+   [88]],
+  ["data:text/plain ,X",
+   "text/plain",
+   [88]],
+  ["data:text/plain%20,X",
+   "text/plain%20",
+   [88]],
+  ["data:text/plain\f,X",
+   "text/plain%0c",
+   [88]],
+  ["data:text/plain%0C,X",
+   "text/plain%0c",
+   [88]],
+  ["data:text/plain;,X",
+   "text/plain",
+   [88]],
+  ["data:;x=x;charset=x,X",
+   "text/plain;x=x;charset=x",
+   [88]],
+  ["data:;x=x,X",
+   "text/plain;x=x",
+   [88]],
+  ["data:text/plain;charset=windows-1252,%C2%B1",
+   "text/plain;charset=windows-1252",
+   [194, 177]],
+  ["data:text/plain;Charset=UTF-8,%C2%B1",
+   "text/plain;charset=UTF-8",
+   [194, 177]],
+  ["data:image/gif,%C2%B1",
+   "image/gif",
+   [194, 177]],
+  ["data:IMAGE/gif,%C2%B1",
+   "image/gif",
+   [194, 177]],
+  ["data:IMAGE/gif;hi=x,%C2%B1",
+   "image/gif;hi=x",
+   [194, 177]],
+  ["data:IMAGE/gif;CHARSET=x,%C2%B1",
+   "image/gif;charset=x",
+   [194, 177]],
+  ["data: ,%FF",
+   "",
+   [255]],
+  ["data:%20,%FF",
+   "",
+   [255]],
+  ["data:\f,%FF",
+   "",
+   [255]],
+  ["data:%1F,%FF",
+   "",
+   [255]],
+  ["data:\u0000,%FF",
+   "",
+   [255]],
+  ["data:%00,%FF",
+   "",
+   [255]],
+  ["data:text/html  ,X",
+   "text/html",
+   [88]],
+  ["data:†,X",
+   "",
+   [88]],
+  ["data:†/†,X",
+   "%e2%80%a0/%e2%80%a0",
+   [88]],
+  ["data:X,X",
+   "",
+   [88]],
+  ["data:image/png,X X",
+   "image/png",
+   [88, 32, 88]],
+  ["data:application/xml,X X",
+   "application/xml",
+   [88, 32, 88]],
+  ["data:unknown/unknown,X X",
+   "unknown/unknown",
+   [88, 32, 88]],
+  ["data:text/plain;a=\",\",X",
+   "data:text/plain;a=\"\\\"\"",
```

Comment:
Is this correct?

Reply:
So you split on the first comma. That means `"` is the value of the `a` parameter. If you serialize that the value becomes `"\""`. Add some more slashes because it's JSON and you got the above.

Reply:
I mean, is "data:" needed in the expected value?

Reply:
Aah, no it's not! Thanks. Same below indeed.